src/main/java/ml/shifu/shifu/core/dtrain/wdl/WDLParallelGradient.java

```diff
@@ -143,6 +165,17 @@ public WDLParams doCompute() {
         return params;
     }
 
+    public WDLParams doCompute(int iteration, int trainBatchSize, int validateBatchSize) {
+        if(validateBatchSize < 10) {
+            LOG.info("Origin validationBatchSize is " + validateBatchSize + " which is so less and adjust to a large value");
+            validateBatchSize = Math.min(this.testData.size(), 10);
+        }
+        LOG.info("training on iteration: " + iteration + " with trainBatchSize: " + trainBatchSize + ", validateBatchSize:" + validateBatchSize);
+        adjustTrainSet(iteration * trainBatchSize % this.trainData.size(), trainBatchSize);
+        adjustTestSet(iteration * validateBatchSize % this.testData.size(), validateBatchSize);
```

Comment:
Could we not enable mini batch in validation set? Validation set no need and better to leverage all validation set each time for a better validation error value.

Reply:
Remove minibatch for validation set